### ISSUE TITLE
As a databrowser editor i want to edit the json directly if i have the Role "JsonEditor" assigned

### ISSUE THREAD

**Issue by RudiThoeni** [noi-techpark/opendatahub-databrowser#334]

**Comment by gappc** [comment]
@RudiThoeni I took a look at possible solutions:

- [vue-json-pretty](https://github.com/leezng/vue-json-pretty): already used in the project, has only restricted editing capabilities, meaning that only existing values can be modified (no property renaming, no addition / removal of properties, no addition / removal of array entries)
- [codemirror](https://codemirror.net/): nice editor, adds ~ 150KB (zipped) download volume to the RAW page
- [monaco](https://github.com/microsoft/monaco-editor): the VSCode editor, adds ~1.1MB (zipped) download volume to the RAW page, not mobile firendly

The `vue-json-pretty` has not enough features (very limited set of edit capabilities), so in my opinion it is not the right choice.

`monaco` is a great editor, but it has a huge foodprint (1.1MB zipped) and it is not mobile friendly.

Finally, `codemirror` seems to be the best fit for the purpose, but it also increases the download volume. If this is acceptable, I'd opt for `codemirror`.

@RudiThoeni @sseppi what are your opinions?

**Comment by sseppi** [comment]
Since for Patrick is important that the Data Browser works also on mobile, I agree that `monaco` isn't the right choice.

Probably `codemirror` is the best option, if `vue-json.pretty` hasn't all the edit capabilities we need.

@gappc since we have a lot of work already planned in the Milestone 2.2 (the next one), shall we program the implementation of this functionality in the Milestone 2.3?

**Comment by gappc** [comment]
@sseppi yes, I'd move it to Milestone 2.3 or even later, as this is a nice-to-have functionality that will be used only by a very few users. @RudiThoeni do you agree or do you think we should prioritize it more?

**Comment by RudiThoeni** [comment]
fine for me!

**Comment by sseppi** [comment]
OK, so let keep it in the Milestone 2.3 and then take the final decision at the beginning of the Milestone development when we decide all the functionalities will be implemented in the iteration.

**Comment by sseppi** [comment]
@Xeroxi91 super! Thank you. Also here I think we are ready to implement the features now!
